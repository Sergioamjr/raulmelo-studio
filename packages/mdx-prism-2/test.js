'use strict';

const rehype = require('rehype');
const dedent = require('dedent');
const rehypePrism = require('./index');

const processHtml = html => {
  return rehype()
    .data('settings', { fragment: true })
    .use(rehypePrism)
    .processSync(html)
    .toString();
};

describe('rehypePrism', () => {
  test('finds code and highlights', () => {
    const result = processHtml(dedent`
      <div>
        <p>foo</p>
        <pre><code class="language-css">p { color: red }</code></pre>
      </div>
    `);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block without language- class', () => {
    const result = processHtml(dedent`
      <pre><code>p { color: red }</code></pre>
    `);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block with fake language- class', () => {
    const result = processHtml(dedent`
      <pre><code class="language-thisisnotalanguage">p { color: red }</code></pre>
    `);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block with no-highlight class', () => {
    const result = processHtml(dedent`
      <pre><code class="language-css no-highlight">p { color: red }</code></pre>
    `);
    expect(result).toMatchSnapshot();
  });

  test('does nothing to code block with nohighlight class', () => {
    const result = processHtml(dedent`
      <pre><code class="language-css nohighlight">p { color: red }</code></pre>
    `);
    expect(result).toMatchSnapshot();
  });
});
