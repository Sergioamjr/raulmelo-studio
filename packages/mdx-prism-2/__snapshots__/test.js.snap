// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehypePrism does nothing to code block with fake language- class 1`] = `"<pre><code class=\\"language-thisisnotalanguage\\">p { color: red }</code></pre>"`;

exports[`rehypePrism does nothing to code block with no-highlight class 1`] = `"<pre><code class=\\"language-css no-highlight\\">p { color: red }</code></pre>"`;

exports[`rehypePrism does nothing to code block with nohighlight class 1`] = `"<pre><code class=\\"language-css nohighlight\\">p { color: red }</code></pre>"`;

exports[`rehypePrism does nothing to code block without language- class 1`] = `"<pre><code>p { color: red }</code></pre>"`;

exports[`rehypePrism finds code and highlights 1`] = `
"<div>
  <p>foo</p>
  <pre><code class=\\"language-css\\"><span class=\\"token selector\\">p</span> <span class=\\"token punctuation\\">{</span> <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> red <span class=\\"token punctuation\\">}</span></code></pre>
</div>"
`;
